app/server/utils/gristify.ts

```diff
@@ -115,6 +116,27 @@ export class Gristifier {
     await db.close();
   }
 
+  /**
+   * Run a query on an SQLite file. Pass the result through
+   * standard Grist decoding for BLOBs. This is handy for
+   * reading objects of the wrong type stored as BLOBs to
+   * protect them in a "cell", or for reading action history,
+   * e.g.
+   *  select * from _gristsys_ActionHistory order by actionNum
+   * This will do the same it would from the sqlite3 utility,
+   * except BLOBs will be expanded.
+   */
+  public async query(queryString: string) {
+    const db = await SQLiteDB.openDBRaw(this._filename);
```

Comment:
If the command is only meant to read data (that's what I understand when I read the command description), what do you think of also passing `mode.OPEN_READING`? I would feel that more securing.